### PR TITLE
fix(kit): accept dates and segments separator as valid input

### LINKED ISSUE #87: 🐞 - `Date` should accept segment-separator as valid input
### Which package(s) are the source of the bug?

@maskito/kit

### Description

1. Open [Date](https://tinkoff.github.io/maskito/kit/date/API) or [DateRange](https://tinkoff.github.io/maskito/kit/date-range).
2. Type `30`
3. Type `.` (default date-segments separator)

**Actual behaviour:**
Nothing happens

**Expected behaviour:**
`30.|` (dot is inserted, cursor after the dot)

**Additional requirements:**
- [ ] Add cypress test

### Maskito version

0.0.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #87 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/kit/src/lib/processors/min-max-date-postprocessor.ts [hunk 1]
```diff
@@ -23,6 +23,13 @@ export function createMinMaxDatePostprocessor({
     datesSeparator?: string;
 }): NonNullable<MaskitoOptions['postprocessor']> {
     return ({value, selection}) => {
+        if (datesSeparator && value.endsWith(datesSeparator)) {
```
Comment (nsbarsukov) [inline projects/kit/src/lib/processors/min-max-date-postprocessor.ts:26]: 1. Open `/maskito/kit/date-range/API?min=2025-05-10`
2. Drop `06.02.2023 –` inside input

File: projects/kit/src/lib/processors/valid-date-preprocessor.ts [hunk 2]
```diff
@@ -12,13 +13,25 @@ export function createValidDatePreprocessor({
     datesSeparator?: string;
 }): NonNullable<MaskitoOptions['preprocessor']> {
     return ({elementState, data}) => {
+        const {value, selection} = elementState;
+
+        if (data === dateSegmentsSeparator && value.length === selection[0]) {
```
Comment (nsbarsukov) [inline projects/kit/src/lib/processors/valid-date-preprocessor.ts:18]: for which cases we need this check `value.length === selection[0]`?

Comment (vladimirpotekhin) [reply projects/kit/src/lib/processors/valid-date-preprocessor.ts:18]: the problem is only at the end of the value, in the middle of the value the mask successfully moves the caret

File: projects/kit/src/lib/processors/valid-date-preprocessor.ts [hunk 3]
```diff
@@ -12,13 +13,25 @@ export function createValidDatePreprocessor({
...
+            return {
+                elementState,
+                data,
+            };
+        }
+
+        if (POSSIBLE_DATES_SEPARATOR.includes(data)) {
```
[the remark this answers is not in the document]
Comment (vladimirpotekhin) [reply projects/kit/src/lib/processors/valid-date-preprocessor.ts:25]: this bug is related to min/max post processor, fixed. But there are still problems with the paste/drop event in the date range. Will be resolved in a separate PR

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**